### ISSUE TITLE
Nil Pointer Dereference when using Cells chat

### ISSUE THREAD

**Issue by autoize** [pydio/cells#242]
In Pydio Cells 2.0.8 there is a "nil pointer dereference" when using Cells chat. It is almost certainly related to [this issue](https://github.com/etcd-io/bbolt/issues/213) with the vendor dependency etcd.io/bbolt which you are using a version from Jan 2019.
 
The stack trace from Pydio matches the issue reported in the other repo exactly, down to the file and line number (84), so it seems to be a race condition - i.e. intermittent. This is why we can sometimes install Cells (on the same VPS) with chat working, and sometimes not.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x2a34591]

goroutine 8280 [running]:
github.com/pydio/cells/vendor/github.com/etcd-io/bbolt.(*Bucket).Cursor(...)
	/opt/teamcity/agent/work/fb9e7e7133d45375/go/src/github.com/pydio/cells/vendor/github.com/etcd-io/bbolt/bucket.go:84
github.com/pydio/cells/broker/chat.(*boltdbimpl).RoomByUuid.func1(0xc004669ce0, 0x0, 0xc004669ce0)
	/opt/teamcity/agent/work/fb9e7e7133d45375/go/src/github.com/pydio/cells/broker/chat/bolt.go:224 +0x91
github.com/pydio/cells/vendor/github.com/etcd-io/bbolt.(*DB).View(0xc000f48e00, 0xc002ffeeb8, 0x0, 0x0)
	/opt/teamcity/agent/work/fb9e7e7133d45375/go/src/github.com/pydio/cells/vendor/github.com/etcd-io/bbolt/db.go:719 +0xa8
github.com/pydio/cells/broker/chat.(*boltdbimpl).RoomByUuid(0xc000c5faa0, 0x3, 0xc004900180, 0x24, 0xc0005983c0, 0x0, 0x0)
	/opt/teamcity/agent/work/fb9e7e7133d45375/go/src/github.com/pydio/cells/broker/chat/bolt.go:222 +0xf5
github.com/pydio/cells/broker/chat/grpc.(*ChatHandler).PostMessage.func1(0xc0022db2a0, 0x7f1bf05c3f48, 0xc000c5faa0)
	/opt/teamcity/agent/work/fb9e7e7133d45375/go/src/github.com/pydio/cells/broker/chat/grpc/handler.go:129 +0x263
created by github.com/pydio/cells/broker/chat/grpc.(*ChatHandler).PostMessage
	/opt/teamcity/agent/work/fb9e7e7133d45375/go/src/github.com/pydio/cells/broker/chat/grpc/handler.go:120 +0x371
```

This occurs regardless of using the Cells official build from Pydio.com, or building it using Go 1.14.x, 1.13.x, or 1.12.x. We attempted to build Cells using a root, and a non-root use, ensuring that Packr is installed. The binary from the non-root user works, with the exception of chat.

We even tried performing the build using a Xeon(R) Gold 6140 vs Xeon(R) CPU E5-2650 VPS, as chat in the binaries built on the Xeon Gold seemed to work intermittently, but the binaries built on the E5 would never work.

Therefore, our conclusion is that this is a race condition bug with Pydio because it uses an old version of the etcd.io/bbolt dependency. Can you please update your etcd.io/bbolt version in an upcoming release to avoid this race condition?

This issue was also reported some weeks ago in the Pydio forum.
https://forum.pydio.com/t/pydio-cells-v2-chat-throws-runtime-error-panic/3271

Thanks in advance for your time.

**Comment by cdujeu** [comment]
Thanks so much for the detailed report! Tried to reproduce it but had no clue how. We’ll keep you posted!

**Comment by cdujeu** [comment]
Hey, 
just pushed the updated vendor on the master, could you kindly try to rebuild from the git and see if it fixes the issue? 
Thx

**Comment by cdujeu** [comment]
Actually it's more likely a bug in the code (just pushed a fix for that)

**Comment by cdujeu** [comment]
Fixed in 79c33c0831d5dd801a10369bb14321030a71862e,  will be 2.1

**Comment by bsinou-agent** [comment]
We have also fixed it on the hotfix branch and shipped it in the 2.0.9 that we shipped yesterday (will be officialy out on Monday...).
Let us know if it is good for you now.
